### PR TITLE
add dirNameRoutePrefix option

### PR DESCRIPTION
**Pull request by davidmarkclements**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ function getScriptType (fname, packageType) {
 }
 
 async function findPlugins (dir, options, accumulator = [], prefix) {
-  const { indexPattern, ignorePattern, scriptPattern } = options
+  const { indexPattern, ignorePattern, scriptPattern, dirNameRoutePrefix = true } = options
```
Comment (jsumners) [inline index.js:60]: Shouldn't this default to `false`?

Comment (davidmarkclements) [reply index.js:60]: the default behaviour is currently true - explicitly setting the option to false restore the prior behaviour of *not* doing this

Comment (jsumners) [reply index.js:60]: /me goes back to knowing nothing of this module.

Comment (davidmarkclements) [reply index.js:60]: hahahah - the behaviour changed from v2 to v3 to automatically being named after the folder. It's a cool feature but it needs an opt-out imo, especially for migration purposes

**Review by fox1t — APPROVED** [review]

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Pull request merged**